### ISSUE TITLE
Query timeout and cancellation

### ISSUE THREAD

**Issue by snaga** [snaga/Hecatoncheir#3]
User story:
* As a system admin, I want long-running (and heavy) queries to be detected and be cancelled automatically, so that I can avoid heavy (and long) loads on my data warehouse system.

Acceptance Criteria:
* `db-run-profiler` has `--timeout` option to cancel queries that exceed the timeout limit.
* Each db driver has the timeout parameter and can cancel queries that exceed the timeout limit.
* The code has tests and passes those tests.
* The code passes pep8 style checks.
* Docs (README and user manual) have been updated.
* All the modifications have been merged to the develop branch.

**Comment by snaga** [comment]
How to detect timeout and cancel queries in each database:

* PostgreSQL
    * Issue `set statement_timeout` before starting a query.
    * Catch the exception caused by the timeout.
    * Raise the QueryTimeout exception from the driver.
* MySQL
    * Get the connection id with `SELECT CONNECTION_ID( )` before starting a query.
    * Start a timer thread (`threading.Timer`) with the cancel callback function.
    * When the query exceeds the timeout limit, connect to the server with another connection, and issue `KILL <PID>` to cancel the query.
    * Reconnect to the MySQL server.
    * Raise the QueryTimeout exception from the driver.
* Oracle
    * Start a timer thread (`threading.Timer`) with the cancel callback function.
    * When the query exceeds the timeout limit, and call `Connection.cancel()` to cancel the query.
        * http://cx-oracle.readthedocs.io/en/latest/connection.html#Connection.cancel
    * Catch the exception caused by the user cancel.
    * Raise the QueryTimeout exception from the driver.
* SQL Server
    * Get the session id with `SELECT @@SPID` before starting a query.
    * Start a timer thread (`threading.Timer`) with the cancel callback function.
    * When the query exceeds the timeout limit, connect to the server with another connection, and issue `KILL <session_id>` to cancel the query.
    * Raise the QueryTimeout exception from the driver.

**Comment by iwanaga** [comment]
:+1:

**Comment by snaga** [comment]
Merged. #8

**Comment by snaga** [comment]
As of today, the query timeout and cancellation has not worked on SQL Server, but it can work on other databases. So, this issue can be closed with opening a new issue on SQL Server. #10
